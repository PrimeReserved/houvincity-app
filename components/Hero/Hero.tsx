"use client"

import React from "react";
import StyledHeading from "../Typography/StyledHeading";
import StyledText from "../Typography/StyledText";
import IHeroProps from "@/interfaces/IHeroProps"



const Hero: React.FC<IHeroProps> = ({ image, title, description, children }) => {
  const backgroundImageStyle = {
    backgroundImage: image ? `url(${image})` : `url('./images/blog/Hero section.png')`,
  };

  console.log(backgroundImageStyle)

  return (
    <div className="hero min-h-[20rem] flex items-center justify-center" 
    data-testid="hero" 
    data-background-image={backgroundImageStyle.backgroundImage}
    style={backgroundImageStyle}>
      <div className="hero-content text-center">
        <div className="max-w-md md:max-w-[48rem]">
          {title && <StyledHeading>{title}</StyledHeading>}
          {description && <StyledText>{description}</StyledText>}
          {children}
        </div>
      </div>
    </div>
  );
};

export default Hero;
