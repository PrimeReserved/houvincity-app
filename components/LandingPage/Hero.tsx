import { Suspense } from 'react';
import Image from 'next/image';
import IconButtonHref from '../Buttons/IconButtonHref';
import SearchHomePage from './ SearchHomePage';
import Loading from '@/app/loading';

const Hero: React.FC = () => {


  return (
    <section className="relative w-full mt-[6.5rem] overflow-hidden bg-center bg-cover py-20 md:py-28 " >
      {/* Background Image */}
      <Suspense fallback={<Loading />}>
        <Image
          src="https://res.cloudinary.com/dzd51q99i/image/upload/v1716402146/houvincity/landing-page/Rectangle_23803_jhw5qg.png"
          alt="Hero Image"
          fill
          className="absolute inset-0"
          priority
        />
      </Suspense>

      {/* Overlay */}
      <div className="absolute inset-0 bg-black opacity-50"></div>

      {/* Content */}
      <div className="container mx-auto px-4 pt-10 text-white relative z-10">
        <div className="max-w-2xl">
          {/* Heading */}
          <h1 className="mb-4 text-3xl font-extrabold md:text-4xl xl:text-5xl text-left">
            Unlock Your Dream
            <br />
            Home Now!
          </h1>
          {/* Paragraph */}
          <p className="mb-6 text-lg leading-relaxed md:text-xl text-left">
            PrimeReservedâ€”Your Ultimate Destination for Outstanding Website
            Designs and Seamless Application Developments. Your satisfaction is
            our prime reserve!
          </p>
          {/* Contact Button */}
          <IconButtonHref
            text="Find Property"
            href="/property"
          />
        </div>
      </div>
      <SearchHomePage
       />
    </section>
  );
};

export default Hero;