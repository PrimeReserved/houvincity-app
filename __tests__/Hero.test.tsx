import { render } from '@testing-library/react';
import Hero from '@/components/Hero/Hero';

describe('Hero Component', () => {
  it('renders with provided props', () => {
    const props = {
      image: './images/blog/Hero section.png',
      title: 'Sample Title',
      description: 'Sample Description',
    };

    const { getByTestId, getByText } = render(<Hero {...props} />);
    
    const heroElement = getByTestId('hero');
    expect(heroElement).toBeInTheDocument();

    const titleElement = getByText('Sample Title');
    expect(titleElement).toBeInTheDocument();

    const descriptionElement = getByText('Sample Description');
    expect(descriptionElement).toBeInTheDocument();
  });

  it('renders default image if image prop is not provided', () => {
    const props = {
      title: 'Sample Title',
      description: 'Sample Description',
    };

    const { getByTestId } = render(<Hero {...props} />);
    
    const heroElement = getByTestId('hero');
    expect(heroElement).toBeInTheDocument();

    const backgroundImageStyle = heroElement.getAttribute('data-background-image');
    expect(backgroundImageStyle).toContain(`url('./images/blog/Hero section.png')`);
  });

  it('renders children', () => {
    const props = {
      title: 'Sample Title',
      description: 'Sample Description',
    };

    const { getByText } = render(
      <Hero {...props}>
        <div>Child Component</div>
      </Hero>
    );

    const childElement = getByText('Child Component');
    expect(childElement).toBeInTheDocument();
  });

  it('matches snapshot with provided props', () => {
    const props = {
      image: './images/blog/Hero section.png',
      title: 'Sample Title',
      description: 'Sample Description',
    };

    const { asFragment } = render(<Hero {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot with default image if image prop is not provided', () => {
    const props = {
      title: 'Sample Title',
      description: 'Sample Description',
    };

    const { asFragment } = render(<Hero {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot with children', () => {
    const props = {
      title: 'Sample Title',
      description: 'Sample Description',
    };

    const { asFragment } = render(
      <Hero {...props}>
        <div>Child Component</div>
      </Hero>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
