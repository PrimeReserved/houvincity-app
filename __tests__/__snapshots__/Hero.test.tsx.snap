// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Component matches snapshot with children 1`] = `
<DocumentFragment>
  <div
    class="hero min-h-[20rem] flex items-center justify-center"
    data-background-image="url('./images/blog/Hero section.png')"
    data-testid="hero"
  >
    <div
      class="hero-content text-center"
    >
      <div
        class="max-w-md md:max-w-[48rem]"
      >
        <h1
          class="mb-7 text-white text-4xl md:text-5xl font-semibold"
        >
          Sample Title
        </h1>
        <p
          class="mt-5 text-heroText text-[17px] px-8 font-medium"
        >
          Sample Description
        </p>
        <div>
          Child Component
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Hero Component matches snapshot with default image if image prop is not provided 1`] = `
<DocumentFragment>
  <div
    class="hero min-h-[20rem] flex items-center justify-center"
    data-background-image="url('./images/blog/Hero section.png')"
    data-testid="hero"
  >
    <div
      class="hero-content text-center"
    >
      <div
        class="max-w-md md:max-w-[48rem]"
      >
        <h1
          class="mb-7 text-white text-4xl md:text-5xl font-semibold"
        >
          Sample Title
        </h1>
        <p
          class="mt-5 text-heroText text-[17px] px-8 font-medium"
        >
          Sample Description
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Hero Component matches snapshot with provided props 1`] = `
<DocumentFragment>
  <div
    class="hero min-h-[20rem] flex items-center justify-center"
    data-background-image="url(./images/blog/Hero section.png)"
    data-testid="hero"
  >
    <div
      class="hero-content text-center"
    >
      <div
        class="max-w-md md:max-w-[48rem]"
      >
        <h1
          class="mb-7 text-white text-4xl md:text-5xl font-semibold"
        >
          Sample Title
        </h1>
        <p
          class="mt-5 text-heroText text-[17px] px-8 font-medium"
        >
          Sample Description
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
